tests/Behat/Context/Setup/SubscriptionContext.php

```diff
@@ -45,8 +102,56 @@ public function thisOrderHasAnActiveMollieSubscription(OrderInterface $order): v
         $subscriptionConfiguration = $subscription->getSubscriptionConfiguration();
         $subscriptionConfiguration->setInterval('10 days');
 
+        $orderItem = $order->getItems()->first();
+        Assert::assertInstanceOf(OrderItemInterface::class, $orderItem);
+        $subscription->setOrderItem($orderItem);
+        $subscription->setPaymentState(MollieSubscriptionInterface::PAYMENT_STATE_OK);
+        $subscriptionConfiguration = $subscription->getSubscriptionConfiguration();
+        $subscriptionConfiguration->setInterval('1 days');
+        $subscriptionConfiguration->setNumberOfRepetitions(3);
+        $subscriptionConfiguration->setPaymentDetailsConfiguration($paymentDetails);
+
         $this->entityManagerSubscription->persist($subscription);
         $this->entityManagerSubscription->persist($subscriptionConfiguration);
+        $this->entityManagerSubscription->persist($mollieCustomer);
+        $this->entityManagerSubscription->flush();
+
+        $this->sharedStorage->set('subscription', $subscription);
+    }
+
+    /**
+     * @Given /^(this subscription) has an active schedule$/
+     */
+    public function thisSubscriptionHasAnActiveSchedule(MollieSubscriptionInterface $subscription): void
+    {
+        $this->subscriptionScheduleProcessor
+            ->processScheduleGeneration($subscription);
+
+        /** @var MollieSubscriptionScheduleInterface $schedule */
+        foreach ($subscription->getSchedules() as $schedule) {
+            $schedule->setScheduledDate(new \DateTime());
+        }
+
+        $this->entityManagerSubscription->persist($subscription);
         $this->entityManagerSubscription->flush();
     }
+
+    /**
+     * @When I run command :command
+     */
+    public function iRunCommand(string $command): void
+    {
+        $this->mollieApiClient->setApiEndpoint('http://localhost:8218');
```

Comment:
I think it's unnecessary as we have separate api client service for test environment

Reply:
Sure, removed